.github/workflows/PR.yml

```diff
@@ -0,0 +1,38 @@
+name: Pinecone CI
+
+on: 
+  pull_request: {}
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  run-tests:
+    name: Run integration tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.20.0
+      - name: Cache NPM dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('package-lock.json') }}
+      - name: Install npm packages
+        run: | 
+          npm install --ignore-scripts
+      - name: Build typescript
+        run: |
+          npm run build
+      - name: Run tests
+        env:
+          CI: true
+          PINECONE_API_KEY: ${{ secrets.PINECONE_API_KEY }}
+          PINECONE_ENVIRONMENT: ${{ secrets.PINECONE_ENVIRONMENT }}
```

Comment:
Which environment do you plan to run this in?

Reply:
At the moment I just have them running in my personal free tier env, but need to sort out something else soon. Any suggestion?

Reply:
We should probably discuss this somewhere less public.